src/Adapter/ImageType/ImageTypeDataProvider.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\ImageType;
+
+use PrestaShop\PrestaShop\Core\ImageType\ImageTypeDataProviderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Get image type categories.
+ */
+class ImageTypeDataProvider implements ImageTypeDataProviderInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getImageTypeCategories(): array
+    {
+        return [
```

Comment:
I think this should be like this is currently, because if we hard-code it, this will not be able to handle different languages, I think. Same for the other comment.

Reply:
Maybe you can populate the property in the constructor?

Reply:
@atomiix I don't think it is possible. As far as I understand Symfony cache, it caches instances and then translations would be again hard-coded.